DOCSP-12627: [MONGOSH] Transaction, Session, and Replica Set Config methods (#67)
mongodb/docs-mongodb-shell: source/reference/methods.txt

       :manual:`plan cache entries </core/query-plans/>` 
       for a collection.

Replication Methods 
-------------------

.. list-table::
   :widths: 30 70
   :header-rows: 1

   * - Method

     - Description

   * - :method:`rs.add()`

     - Adds a member to the replica set. You must connect to the 
       primary of the replica set to run this method.

   * - :method:`rs.addArb()`

     - Adds an arbiter to an existing replica set.

   * - :method:`rs.config()`

     - Returns a document that contains the current replica set configuration.

   * - :method:`rs.freeze()`

     - Makes the replica set member that ``mongosh`` is connected to 

>        ineligible to become primary for the specified duration. The 
>        duration must be specified in seconds.

Reviewer: active voice
```suggestion
       ineligible to become primary for the specified duration. You 
       must specify the duration in seconds.
```

Author: 👍

   * - :method:`db.getReplicationInfo()`

     - Returns the status of the replica set from the oplog data.

   * - :method:`rs.initiate()`

     - Initializes a new replica set.

   * - ``rs.isMaster``

     - Returns the replica set configuration, role, and status for 

>        the ``mongod`` instance that ``mongosh`` is connected to. 

Reviewer: adds a link (if this is configured for this repository)
```suggestion
       the |mongod| instance that ``mongosh`` is connected to. 
```

Author: 👍

>        The ``mongosh`` ``rs.isMaster()`` method wraps the server :dbcommand:`isMaster` command.

Reviewer: `rs.isMaster()` here has parentheses. The reference above does not. Think they should be consistent. Not sure which one is correct.

Author: With parens is correct, I believe.

   * - :method:`db.printReplicationInfo()`

     - Returns the oplog of the replica set member that ``mongosh`` is 
       connected to. 
       

>        This is identical to the :method:`rs.printReplicationInfo()` method.

Reviewer: These statements are not used consistently: `This is identical to ...`. If the same applies for methods listed above and below, consider adding these phrases each of those.

An alternative option is to remove these statements. I can see readers viewing them as redundant.

Author: I didn't actually write up these particular methods (just moved them around to put them in alphabetical order) so I'm not sure what the original intent was, but it seems reasonable to remove the "identical" sentences.

       :method:`db.printSlaveReplicationInfo()` method. The :method:`db.printSlaveReplicationInfo()` method is deprecated in ``mongosh``. Use 
       ``db.printSecondaryReplicationInfo()`` instead.

       The following is an example output from the 
       ``rs.printSecondaryReplicationInfo()`` method issued on a replica set with two secondary members:

       .. code-block:: sh 
          :copyable: false 

          source: rs2.example.net:27017
          {
            syncedTo: 'Tue Oct 13 2020  09:37:28 GMT-0700 (Pacific Daylight Time)',
            replLag: '0 secs (0 hrs) behind the primary '
          }
          ---
          source: rs3.example.net:27017
          {
            syncedTo: 'Tue Oct 13 2020  09:37:28 GMT-0700 (Pacific Daylight Time)',
            replLag: '0 secs (0 hrs) behind the primary '
          }

   * - ``rs.printSecondaryReplicationInfo``

     - Returns the status of the secondary members of the replica set. 
     
       This is identical to the ``db.printSecondaryReplicationInfo()`` 
       method. This method's output is similar to the 
       :method:`rs.printSlaveReplicationInfo()` method in the legacy 

>        :binary:`~mongo` shell. The :method:`rs.printSlaveReplicationInfo()` method is deprecated in ``mongosh``. Use 

Reviewer: if this replacement is configured for this repo.
```suggestion
       |mongo| shell. The :method:`rs.printSlaveReplicationInfo()` method is deprecated in ``mongosh``. Use 
```

Author: 👍
